lib/OpenQA/Markdown.pm

```diff
@@ -25,30 +19,47 @@ sub is_light_color {
     return $sum > 380;
 }
 
-sub _bugref_to_html ($bugref) {
-    my $bugurl = bugurl($bugref);
-    return "<a href=\"$bugurl\">$bugref</a>";
+sub bugref_to_html ($bugref, $fancy = 0) {
+    my $app = OpenQA::App->singleton;
+    my $bugs = $app->schema->resultset('Bugs');
+    my $bug = $bugs->get_bug($bugref);
+    my $bugurl = $app->bugurl_for($bugref);
+    my $bugtitle = $app->bugtitle_for($bugref, $bug);
+    my $bugicon = $app->bugicon_for($bugref, $bug);
+    if ($fancy) {
+        return
+qq{<span title="$bugtitle" class="openqa-bugref"><a href="$bugurl"><i class="test-label $bugicon"></i>&nbsp;$bugref</a></span>};
+    }
+    return "<a href=\"$bugurl\" title=\"$bugtitle\">$bugref</a>";
```

Comment:
Making a database query is generally expensive. It would be better if we could avoid it in the non-fancy case. I think these should really be two distinct functions. Keep the old function as-is and introduce a new one which does the database lookup.

Reply:
It is also not so nice in general that the Markdown module now depends on database operations. Maybe this new function would also better live at another place, e.g. the database schema (to make it explicit that this is formatting based on information from the database).

Reply:
In both cases information from the database (e.g. the title in the non-fancy case) is shown.
I actually think that this is the right place, as the comment rendering is the logical location for this.
Also it was here before.

Reply:
Really? You even had to introduce `my $t = Test::Mojo->new('OpenQA::WebAPI');` to make the corresponding test work - so I would assume this worked before without Mojo app and database.

And the non-fancy case is really only using the bugref and bugurl variables. Nothing that would require a database lookup. The old code was also just using the free function bugurl to compute the bug URL so maybe that's how it should be kept (and all other places just keep using that simple function directly or via some helper if it needs to be).

Reply:
yes - but now it is used.
The non-fancy case uses `return qq{<a href="$bugurl" title="$bugtitle">$bugref</a>};` - the title is used and comes from the database.
The redundant bugurl function was removed for good - we really shouldn't duplicate that.

Reply:
Just to chime in purely from a user's perspective. I find it invaluable to see the title of a ticket in Redmine, and I often fix links that aren't linked correctly because opening those tickets takes a lot more time.

Reply:
Note that this is also called when retrieving the comments of a job via the API, e.g.
```
% openqa-cli api --o3  jobs/3702970/comments | jq
[
  {
    "bugrefs": [],
    "created": "2023-11-04 23:56:34 +0000",
    "id": 508341,
    "renderedMarkdown": "<p>Automatic investigation jobs...
...
```
osc, qem-bot and our tooling are requesting this route, so a new database query should be carefully considered.
I agree though that seeing the issue title is really nice.
Maybe we can change the api route to not render the markdown at all, because I believe in most cases they are not interested in that. For our tooling I'm sure, we can check for the others. It could be done via an extra query string parameter `renderHTML=1` if someone is really interested to get the HTML.

Reply:
As discussed we should change to *not* deliver "renderedMarkdown" by default but allow with a flag, e.g. `openqa-cli api --o3  jobs/3702970/comments rendered_markdown=1`

Reply:
done